src/feature/relay/dns.c

```diff
@@ -1385,42 +1385,63 @@ configure_nameservers(int force)
   evdns_set_log_fn(evdns_log_cb);
   if (conf_fname) {
     log_debug(LD_FS, "stat()ing %s", conf_fname);
-    if (stat(sandbox_intern_string(conf_fname), &st)) {
-      log_warn(LD_EXIT, "Unable to stat resolver configuration in '%s': %s",
+    int missing_resolv_conf = 0;
+    int stat_res = stat(sandbox_intern_string(conf_fname), &st);
+
+    if (stat_res) {
+      log_warn(LD_EXIT, "unable to stat resolver configuration in '%s': %s",
                conf_fname, strerror(errno));
-      goto err;
-    }
-    if (!force && resolv_conf_fname && !strcmp(conf_fname,resolv_conf_fname)
+      missing_resolv_conf = 1;
+    } else if (!force && resolv_conf_fname &&
+               !strcmp(conf_fname,resolv_conf_fname)
         && st.st_mtime == resolv_conf_mtime) {
-      log_info(LD_EXIT, "No change to '%s'", conf_fname);
+      log_info(LD_EXIT, "no change to '%s'", conf_fname);
       return 0;
     }
+
+    if (st.st_size == 0)
```

Comment:
If the stat() call failed above, then at this point, st.st_size will be uninitialized, and we're not allowed to look at it.

Reply:
Checking if `stat()` succeeded in 770aee4856e9a4a1ee8b338c6ba083117a86ad78.